task/07-yield-tasks.js

```diff
@@ -108,7 +134,22 @@ function* depthTraversalTree(root) {
  *
  */
 function* breadthTraversalTree(root) {
```

Comment:
thre is a way to use javascript array as a queue. See the shift method. Try to refactor this with 1 arrray

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift

Reply:
shift method works too slow, so the solution didn't pass the time limit test